### ISSUE TITLE
'direction TD' in a subgraph results in a parser error

### ISSUE THREAD

**Issue by ferdnyc** [mermaid-js/mermaid#4229]
Inserting `direction TD` into a subgraph causes the graph to fail to render:

```mermaid
flowchart TD
  subgraph SUB1
    A
    B
    C
  end
  subgraph SUB2
    direction LR
    D
    E
  end
  subgraph SUB3
    direction TD
    DB[(Database)]
    F
  end
DB -.- F
SUB1 --> SUB2
SUB3 -.- SUB2
SUB1 --> SUB3
```

`direction TB` works just fine:

```text
flowchart TD
  subgraph SUB1
    A
    B
    C
  end
  subgraph SUB2
    direction LR
    D
    E
  end
  subgraph SUB3
    direction TB
    DB[(Database)]
    F
  end
DB -.- F
SUB1 --> SUB2
SUB3 -.- SUB2
SUB1 --> SUB3
```

```mermaid
flowchart TD
  subgraph SUB1
    A
    B
    C
  end
  subgraph SUB2
    direction LR
    D
    E
  end
  subgraph SUB3
    direction TB
    DB[(Database)]
    F
  end
DB -.- F
SUB1 --> SUB2
SUB3 -.- SUB2
SUB1 --> SUB3
```

_Originally posted by @ferdnyc in https://github.com/mermaid-js/mermaid/issues/2789#issuecomment-1468561170_

**Comment by ruohola** [comment]
It seems that `TD` only works as a direction for `flowchart` or `graph`. Only `direction TB` works inside a `subgraph`.

**Comment by jgreywolf** [comment]
In the documentation TB is specified as one of the direction options.  We will update documentation to make this clearer

**Comment by ferdnyc** [comment]
@jgreywolf 

> In the documentation TB is specified as one of the direction options.

Yes, and `TD` [is specified as a synonym for `TB`](https://mermaid.js.org/syntax/flowchart.html#direction). 

> Possible FlowChart orientations are:
> 
> * TB - Top to bottom
> * TD - Top-down/ same as top to bottom
> * BT - Bottom to top
> * RL - Right to left
> * LR - Left to right

And in `flowchart` declarations, **it is**. It just breaks when used inside a `subgraph` as `direction TD`.

The [section on direction in subgraphs](https://mermaid.js.org/syntax/flowchart.html#direction-in-subgraphs) doesn't make any mention of `TD` not being a valid subgraph direction. It doesn't make any mention of valid direction arguments **at all**. Which would seem to imply that the valid directions are the ones documented for `flowchart`. There's certainly nothing to indicate that `TD` would suddenly _stop_ being a synonym for `TB`.

**Comment by rambda** [comment]
I confirmed that the current behavior is the same as before.

The official documentation still includes `direction TD`.
The `direction TD` syntax is still not implemented in the parser.

A not so serious consequence is that most LLMs, like GPT, Gemini, and Claude all consider `direction TD` to be correct syntax.

**Comment by ferdnyc** [comment]
So, having now read [the parser](https://github.com/mermaid-js/mermaid/blob/8a703bd09fd1f347eeb190db5d5c4ae371bc77dd/packages/mermaid/src/diagrams/flowchart/parser/flow.jison), I know _why_ this doesn't work, at least.

For `flowchart` statements, the parser allows them to be followed by a `<dir>`, which is a direction that can either be any of 10 different strings, or nothing:

https://github.com/mermaid-js/mermaid/blob/8a703bd09fd1f347eeb190db5d5c4ae371bc77dd/packages/mermaid/src/diagrams/flowchart/parser/flow.jison#L127-L137

(The 10 different strings are the familiar TB, LR, RL, BT, the documented TB synonym TD, an undocumented BR, and then any one of four "pointing" characters: `^`, `>`, `v`, or `<`. `flowchart BR` or a character-direction like `flowchart >` does actually work, BTW, despite not being documented.):


```text
flowchart >
  A -.- B
  B --> C
  D -.- E
```

```mermaid
flowchart >
  A -.- B
  B --> C
  D -.- E
```

```text
flowchart BR
  A -.- B
  B --> C
  D -.- E
```

```mermaid
flowchart BR
  A -.- B
  B --> C
  D -.- E
```


When support for `direction` in subgraphs was implemented, they didn't want to accept all of the same 10 `<dir>`s because that would've made something like `direction ^` or `direction >` valid. So they ignored their `<dir>` grammar, and separately implemented parsing for only four `direction` strings:

https://github.com/mermaid-js/mermaid/blob/8a703bd09fd1f347eeb190db5d5c4ae371bc77dd/packages/mermaid/src/diagrams/flowchart/parser/flow.jison#L139-L142

Hence, `direction TD` isn't accepted, nor is `direction BR` (whatever that even means).
